tests/end_to_end/gcc-division/gcc-o2-out.c

```diff
@@ -0,0 +1,373 @@
+    func_00400090(arg0 % 0xF4240);
+    func_00400090(arg0 % 0x989680);
+    func_00400090(arg0 % 0x5F5E100);
+    func_00400090(arg0 % 0x3FFFFFFE);
+    func_00400090(arg0 - ((arg0 / 0x3FFFFFFE) * 0x3FFFFFFF));
+    phi_a0_5 = arg0;
+    if (arg0 < 0) {
+        phi_a0_5 = arg0 + 0x3FFFFFFF;
+    }
+    func_00400090(arg0 - ((phi_a0_5 >> 0x1E) << 0x1E));
+    func_00400090(arg0 - ((arg0 / 0x40000000) * 0x40000001));
+    func_00400090(arg0 - ((arg0 / 0x80000000) * 0x7FFFFFFD));
+    func_00400090(arg0 - ((arg0 / 0x7FFFFFFD) * 0x7FFFFFFE));
+    func_00400090(arg0 - ((arg0 / 0x80000000) * 0x7FFFFFFF));
+    func_00400090(arg0 & 0x7FFFFFFF);
+    func_00400090(arg0 - ((arg0 / 0x80000000) * 0x7FFFFFFF));
+    func_00400090(arg0 - ((arg0 / 0x7FFFFFFD) * 0x7FFFFFFE));
+    func_00400090(arg0 % 0xA);
+    func_00400090(arg0 % 7);
+    func_00400090(arg0 % 5);
+    phi_a0_6 = arg0;
+    if (arg0 < 0) {
+        phi_a0_6 = arg0 + 3;
+    }
+    func_00400090(arg0 - ((phi_a0_6 >> 2) * 4));
+    func_00400090(arg0 % 3);
+    func_00400090(arg0 % 2);
+    func_00400090(0U);
+}
+
+void func_00401AA0(u32 arg0) {
+    s32 temp_hi;
+    s32 temp_hi_2;
+    s32 temp_hi_3;
+    s32 temp_hi_4;
+    s32 temp_hi_5;
+    s32 temp_hi_6;
+
+    func_00400090(arg0);
+    func_00400090(arg0 >> 1);
+    func_00400090(arg0 / 3);
+    func_00400090(arg0 >> 2);
+    func_00400090(arg0 / 5);
+    func_00400090(arg0 / 6);
+    temp_hi = arg0 / 7;
+    func_00400090((u32) (temp_hi + ((u32) (arg0 - temp_hi) >> 1)) >> 2);
+    func_00400090(arg0 >> 3);
+    func_00400090(arg0 / 9);
+    func_00400090(arg0 / 0xA);
+    func_00400090(arg0 / 0xB);
+    func_00400090(arg0 / 0xC);
+    func_00400090(arg0 / 0xD);
+    func_00400090(arg0 / 0xE);
+    func_00400090(arg0 / 0xF);
+    func_00400090(arg0 >> 4);
+    func_00400090(arg0 / 0x11);
+    func_00400090(arg0 / 0x12);
+    temp_hi_2 = MULTU_HI(arg0, 0xAF286BCB);
+    func_00400090((u32) (temp_hi_2 + ((u32) (arg0 - temp_hi_2) >> 1)) >> 4);
+    func_00400090(arg0 / 0x14);
+    temp_hi_3 = MULTU_HI(arg0, 0x86186187);
+    func_00400090((u32) (temp_hi_3 + ((u32) (arg0 - temp_hi_3) >> 1)) >> 4);
+    func_00400090(arg0 / 0x16);
+    func_00400090(arg0 / 0x17);
+    func_00400090(arg0 / 0x18);
+    func_00400090(arg0 / 0x19);
+    func_00400090(arg0 / 0x1A);
+    temp_hi_4 = MULTU_HI(arg0, 0x2F684BDB);
+    func_00400090((u32) (temp_hi_4 + ((u32) (arg0 - temp_hi_4) >> 1)) >> 4);
+    func_00400090(arg0 / 0x1C);
+    func_00400090(arg0 / 0x1D);
+    func_00400090(arg0 / 0x1E);
+    temp_hi_5 = MULTU_HI(arg0, ((s32) &D_8421085 | 0));
+    func_00400090((u32) (temp_hi_5 + ((u32) (arg0 - temp_hi_5) >> 1)) >> 4);
+    func_00400090(arg0 >> 5);
+    func_00400090(arg0 / 0x21);
+    func_00400090(arg0 / 0x64);
+    func_00400090(arg0 / 0xFF);
+    func_00400090(arg0 / 0x168);
+    func_00400090(arg0 / 0x3E8);
+    func_00400090(arg0 / 0x2710);
+    func_00400090((u32) MULTU_HI((arg0 >> 5), ((s32) &D_A7C5AC5 | 0)) >> 7);
```

Comment:
looks like the "add test" script misidentified this as a symbol instead of a constant

Reply:
Yeah - I was going to leave this alone? In case we want to to fix this kind of error in the future? (Or, I could manually edit the `.s` if you don't like that)

Reply:
Depends on if we have a concrete plan on actually fixing the import step to disassemble .o files correctly -- if not I'd prefer to manually fix up the .s. I've been treating the import process as a single-time thing, with e.g. manual fixes to make jump tables work, removing the IDO -g version if not relevant, etc., and I don't think existing tests have these kind of broken import artifacts.

Reply:
Ah - I meant if we want to do something about it on the m2c side. (Warn that the symbol is fake, or try to convert it to a literal)
I'll just edit the .s though.

Reply:
ah, I see! I think it's fine to leave that to manual cleanup, mis-identified literals should be rare given a good disassembly, particularly so combined with division